### PR TITLE
docs(hooks): add hooks doc onInit useDefaultProps and add import

### PR DESCRIPTION
**Pull request by kingzez**
## Description

Add a short description of:
- add hooks doc onInit useDefaultProps and add import 
- fix TOC level

### CODE REVIEW

**Comment by tomByrer** [comment]
Thanks for keeping the docs up to date with progress!

I'm wondering if the examples you've added should also be added to [the fiddle examples](https://github.com/BuilderIO/mitosis/blob/main/packages/fiddle/src/constants/templates/jsx-templates.ts)?  I tend to play with code before reading....

File: docs/hooks.md [hunk 1]
```diff
@@ -137,3 +202,47 @@ export default function OnUpdateWithDeps() {
   return <div />;
 }
 ```
+
+## useDefaultProps
+
+The useDefaultProps hook is set default values for the props.
```
Comment (samijaber) [inline docs/hooks.md:208]:
```suggestion
The `useDefaultProps` hook sets default values for a component's props:
```

File: docs/hooks.md [hunk 2]
```diff
@@ -77,18 +80,76 @@ export default forwardRef(function MyInput(props, inputRef) {
...
+      }}
+      type="button"
+    >
+      Button
+    </button>
+  );
 }
+```
+
+And useStyle can also use outside component:
```
Comment (samijaber) [inline docs/hooks.md:107]:
```suggestion
`useStyle` can also be used outside of the component's body:
```

File: docs/hooks.md [hunk 3]
```diff
@@ -77,18 +80,76 @@ export default forwardRef(function MyInput(props, inputRef) {
...
+    color: white;
+    font-size: 12px;
+    outline: 1px solid black;
+  }
 `);
 ```
 
+## onInit
+
+The onInit hook is the best place to put custom code to execute before the component mounts, it is executed earlier than the onmount hook.
```
Comment (samijaber) [inline docs/hooks.md:128]:
```suggestion
The `onInit` hook is the best place to put custom code to execute before the component mounts. It is executed before the `onMount` hook.
```

**Review by samijaber — APPROVED** [review]
Thank you so much for improving the docs 🙏🏽

**Comment by samijaber** [comment]
Good point @tomByrer: our examples need to improve to showcase all the new features we've been adding. Don't want to hold up this PR though, so we can do this in a followup PR.

**Pull request merged**